pkg/domain/form_control_ref.go

```diff
@@ -1,26 +1,80 @@
 package domain
 
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+const (
+	errNoControlNameProvided = "please provide a form control name"
+	errNoRefProvided         = "please provide a ref"
+	errNoValueProvided       = "please provide a value"
+	errInvalidFormControl    = "invalid form control provided with name [%s]"
+	errInvalidFormControlRef = "unable to validate form control ref with information provided"
+)
+
 type FormControlRefList struct {
 	Count              int32             `json:"count"`
 	FormControlRefList []*FormControlRef `json:"formControlList"`
 }
 
 type FormControlRef struct {
-	Id          int32  `json:"id,omitempty"`
-	Ref         string `json:"ref,omitempty"`
-	Name        string `json:"name"`
-	Enabled     bool   `json:"enabled,omitempty"`
-	Description string `json:"description"`
-	FormControl string `json:"formControl"`
-	SortOrder   int32  `json:"order"`
-	Key         string `json:"key"`
-	Value       string `json:"value"`
-	TargetRef   string `json:"targetRef"`
-	Audit       *Audit `json:"audit,omitempty"`
+	Id            int32       `json:"id,omitempty"`
+	TargetRef     string      `json:"targetRef" validate:"required"`
+	SortOrder     int32       `json:"sortOrder" validate:"required"`
+	Description   string      `json:"description" validate:"required"`
+	Name          string      `json:"name" validate:"required"`
+	Key           string      `json:"key" validate:"required"`
+	Value         string      `json:"value" validate:"required"`
+	FormControlId string      `json:"formControlId" validate:"required"`
+	FormControl   FormControl `json:"formControl,omitempty"`
+	Audit         *Audit      `json:"audit,omitempty"`
+	Ref           string      `json:"ref,omitempty"`
+	Version       int32       `json:"version,omitempty"`
 }
 
 func NewFormControlRef() *FormControlRef {
 	return &FormControlRef{
 		Ref: NewRef(FormControlRefTable),
 	}
 }
+
+func (o *FormControlRef) ValidateCreateRequireFields(formControlName string, ref string, value string) error {
```

Comment:
refactor to create a single fun that takes in a string and an errMessage. this new func could be re-used more readily for other fields